src/analyses/ai.h

```diff
@@ -87,17 +90,23 @@ class ai_baset
     finalize();
   }
 
-  /// Accessing individual domains at particular locations
-  /// (without needing to know what kind of domain or history is used)
-  /// A pointer to a copy as the method should be const and
-  /// there are some non-trivial cases including merging domains, etc.
-  /// Intended for users of the abstract interpreter; don't use internally.
-
-  /// Returns the abstract state before the given instruction
+  /// Get the abstract state before the given instruction, without needing to
+  /// know what kind of domain or history is used. Note: intended for
+  /// users of the abstract interpreter; don't use internally.
```

Comment:
:pick: I think this is a typo in the original, I think it should be 
```suggestion
  /// users of the abstract interpreter; don't use externally.
```

Reply:
I think it means internally. Within the class which is derived from `ait` you should (implement and) use `get_state`, which is protected. Code that uses the class derived from `ait` should use these two functions, which are public. I will update the docs to make that clearer (and check that it's correct with someone who knows more about `ait`.

Reply:
P.S. Good use of the "Insert a suggestion" feature, though.